src/plugins/gui/fiatPluginTypes.ts

```diff
@@ -1,11 +1,11 @@
-import { asValue } from 'cleaners'
+import { asEither, asValue } from 'cleaners'
 import { EdgeAccount } from 'edge-core-js'
 
 import { EdgeTokenId } from '../../types/types'
 import { EnterAmountPoweredBy } from './scenes/EnterAmountScene'
 
-export const asFiatPaymentTypes = asValue('credit', 'applepay', 'googlepay')
-export type FiatPaymentType = 'credit' | 'applepay' | 'googlepay'
+export const asFiatPaymentType = asEither(asValue('credit'), asValue('applepay'), asValue('googlepay'), asValue('iach'))
```

Comment:
Please keep using the multi-parameter form of `asValue`, instead of using `asEither` like this. The `asValue` run-time behavior is much better, since it knows to check all the enum variants, and then print a meaningful error message if the value doesn't match any of them.

Reply:
Also, should "iach" just be "ach"? The payment type is the payment type, and the settlement speed is the settlement speed. Mixing the two concepts seems like it will cause problems.

What if MoonPay does instant transfers for most customers, but then decides to wait a day for certain jurisdictions or risk profiles? Do we need two MoonPay plugins at that point, to handle the different speeds?

I think the payment type should be "ach", and the "instant" part plays into how we prioritize things.

Reply:
Having iACH is based on the fact that discussions with several partners show they are looking to offer both and have iACH be more costly than regular ACH so it's not just settlement time but also price.

Reply:
I switched away from the multi param model because it didn't properly map to a type when using `ReturnType <typeof ...>` The new syntax does map correctly so I don't have to duplicate the type and cleaner

Reply:
After further experiments, it turns out that `asValue` does work correctly, and `asEither` actually guesses the type as `string`, which is too loose.